lib/intelligent_foods/api_client.rb

```diff
@@ -2,61 +2,43 @@
 
 module IntelligentFoods
   class ApiClient
-    attr_accessor :access_token
-    attr_reader :id, :secret
-
-    def initialize(id: nil, secret: nil, client: nil)
-      @id = id
-      @secret = secret
+    def initialize(client: nil)
       @client = client || Net::HTTP
     end
 
-    def authenticate!
-      uri = URI("#{IntelligentFoods.base_url}/token")
-      body = { client_id: id, client_secret: secret }
-      request = build_request_with_body(uri: uri, body: body)
-      authorization = IntelligentFoods::Authorization::Basic.
-                      factory(client_id: id, client_secret: secret)
-      response = execute_request(request: request, uri: request.uri,
-                                 authorization: authorization)
-      handle_authentication_response(response: response.data)
-      self
-    end
-
-    def post(path:, body:)
+    def post(path:, body:, authorization: nil)
```

Comment:
i don't think the authorization should be passed into every method call. it should be help be the ApiClient and you could have 2 instance of the client for each of v1 and v2... something is still bugging me about this